diff --git a/snippets/rst.snippets b/snippets/rst.snippets
index 97845de..3eb0f86 100644
--- a/snippets/rst.snippets
+++ b/snippets/rst.snippets
@@ -19,4 +19,22 @@ snippet -
 	${3}
 snippet cont:
 	.. contents::
-	
+snippet code
+	.. code::
+
+		${1:code}
+
+		$0
+snippet cb
+	.. code-block:: ${1:python}
+
+		${2:code}
+
+		$0
+snippet toc
+	.. toctree::
+		:maxdepth: ${1:2}
+snippet ref
+	\`${1:ref} <${2:url}>\`_
+snippet v
+	\`\`${1:variable}\`\`
